script_test.go

```diff
@@ -1219,6 +1221,19 @@ func TestExecRunsGoHelpAndGetsUsageMessage(t *testing.T) {
 	}
 }
 
+func TestWithContextTimeout(t *testing.T) {
+	t.Parallel()
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Nanosecond)
+	defer cancel()
+	p := script.NewPipe().WithContext(ctx).Exec("sleep 1")
```

Comment:
Ideally we wouldn't have to depend on another external command—it makes it hard to run CI tests in scratch containers, for example.

I wondered if we could just write a `Filter` function that waits on the ctx, so that we can test it gets cancelled correctly. But of course `Filter` doesn't _take_ the context.

But perhaps it should! What do you think? I mean, every `Filter` is a potentially long-running concurrent task that could maybe be cancelled. What if there were a `FilterContext`, for example?